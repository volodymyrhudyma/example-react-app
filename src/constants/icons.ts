import { IIcons } from 'src/types';

const icons: IIcons = {
  logo: '/icons/logo.png',
  dashboard: '/icons/dashboard.svg',
  sessions: '/icons/sessions.svg',
  meetings: '/icons/meetings.svg',
  files: '/icons/files.svg',
  conversations: '/icons/conversations.svg',
  settings: '/icons/settings.svg',
  search: '/icons/search.png',
  headerAvatar: '/icons/avatar-header.png',
  articleAvatar: '/icons/avatar-article.png',
  more: '/icons/more.svg',
  delete: '/icons/delete.svg',
  card: '/icons/card.svg',
  cardActive: '/icons/card-active.svg',
  table: '/icons/table.svg',
  tableActive: '/icons/table-active.svg',
};

export default icons;
