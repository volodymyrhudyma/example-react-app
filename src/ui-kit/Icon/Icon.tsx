import React from 'react';
import { Wrapper } from './styled/Icon.styled';

export interface IIconProps {
  src: string;
  alt: string;
  width: number;
  height: number;
}

const Icon = (props: IIconProps) => <Wrapper {...props} />;

Icon.defaultProps = {
  alt: 'Icon',
  width: 24,
  height: 24,
};

export default Icon;
