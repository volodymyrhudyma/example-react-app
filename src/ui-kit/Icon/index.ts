import Component from './Icon';

export default Component;
