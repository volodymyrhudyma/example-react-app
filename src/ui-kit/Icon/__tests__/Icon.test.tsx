import React from 'react';
import { shallow } from 'enzyme';
import icons from 'src/constants/icons';
import Icon from '../Icon';

describe('Icon component', () => {
  it('should render correctly', () => {
    const component = shallow(<Icon src={icons.settings} />);
    expect(component).toMatchSnapshot();
  });
});
