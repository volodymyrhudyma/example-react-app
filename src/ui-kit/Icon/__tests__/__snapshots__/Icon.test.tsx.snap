// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should render correctly 1`] = `
<styled.img
  alt="Icon"
  height={24}
  src="/icons/settings.svg"
  width={24}
/>
`;
